scout/server/blueprints/cases/views.py

```diff
@@ -710,6 +713,10 @@ def delivery_report(institute_id, case_name):
                 ),
                 "warning",
             )
+            LOG.error(
+                f"An error occurred while downloading delivery report {delivery_report} -- {ex}"
+            )
+            return redirect(request.referrer)
```

Comment:
Hm, not quite; this dls the html page for the pdf endpoint, as the A HREF is marked DOWNLOAD in the sidebar!
![Screenshot 2022-02-03 at 09 04 51](https://user-images.githubusercontent.com/758570/152304261-15de742f-10cc-47a5-baee-114752316906.png)

Reply:
Ah, so it looks like it downloads anyway. I'll check!

Reply:
This problem is caused by the download attribute in this line ---> https://github.com/Clinical-Genomics/scout/blob/11c8be6d40128d29d524945abe5a2f42ec892f34/scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html#L521

We put it there some time ago because otherwise the page spinner continued to spin even after the download of the file. I'll try to see if there is a simple way to fix it!

Reply:
One solution would be, while populating the case data to be shown on the case page, check which resource files (delivery reporr, cancer qc report, etx) do actually exist and if they don't exist do not show the at all in the sidebar. This would be the cleanest solution I think. But perhaps in another PR?

Reply:
Yes, that would be nice, and agree on other PR! But it could also change dynamically, as when we e.g. have a mount issue or case update, so we kind of need another way. Perhaps we should just like `abort(404)` instead? I want to avoid a situation where a user thinks they have downloaded the report, uploading it to the LIMS without having checked it. They should perhaps react to the change of file suffix, but not all win users show those.

Reply:
Unless you remove that "download" property from the link tag, it will always download a failed page, no matter how you handle the error, try. Now you are seeing this problem a lot because most cases in scout stage, unlike scout prod,  don't have a delivery report

Reply:
I honestly don't know how to fix it other than removing the "download" but also removing the loading spinner

Reply:
Right, a downloaded 404 is useless. I can't remember why it wasn't possible to just stop the spinner on completion/error with a download script on the case sidebar?

Reply:
https://github.com/Clinical-Genomics/scout/issues/2806

Reply:
I think we should merge this one and focus on the missing files problem somewhere else. Because you could perhaps even check if a file is existing in jinja (before printing "download") but that would be an ugly solution, since the whole logic should be in the view.. And this PR is not actually introducing the error, because it's already like this in prod

Reply:
Would unbinding beforeunload work here as well? A small script on the sidebar would probably as well. But, fine, there is a flash and the file type changes after download. Lets just add an issue for this and merge?

Reply:
> Lets just add an issue for this and merge?

I've done it, we can fix this problem before the next release, so the problem before won't show!

Reply:
Hihi, but as you very correctly identify it was a preexisting problem! You only changed it slightly, but a similar situation existed before.